src/transport/grpc/mod.rs

```diff
@@ -622,13 +640,257 @@ impl<'a> TryFrom<&'a Interfaces> for NodeData {
     }
 }
 
+/// Store wrapper designed specifically for the grpc connection
+/// Used mainly to request device owned properties to the message hub instead of looking them up in the local storage
+// TODO avoid also storing device owned properties if possible
+#[derive(Debug, Clone)]
+pub struct GrpcStore<S> {
+    pub(crate) store: S,
+    pub(crate) client: Arc<Mutex<MsgHubClient>>,
+}
+
+impl<S> GrpcStore<S> {
+    pub(crate) fn new(store: S, client: MsgHubClient) -> Self {
+        Self {
+            store,
+            client: Arc::new(Mutex::new(client)),
+        }
+    }
+}
+
+/// Error returned while operating on the store of a grpc connection
+/// This store needs to request properties from the message hub server
+/// and has additional errors consequently
+#[non_exhaustive]
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum GrpcStoreError<E> {
+    /// Underlying store error
+    #[error("Underlying store error: {0}")]
+    StoreError(E),
+    /// Error while retrieving data from the message hub server
+    #[error("Error while retrieving data from the message hub server: {0}")]
+    StatusError(tonic::Status),
+    /// Error while converting a proto received value to an internal type
+    #[error("Error while converting a proto received value to an internal type: {0}")]
+    ConversionError(MessageHubProtoError),
+}
+
+impl<E> GrpcStoreError<E> {
+    fn from_err(err: E) -> Self {
+        Self::StoreError(err)
+    }
+
+    fn from_status(status: tonic::Status) -> Self {
+        Self::StatusError(status)
+    }
+
+    fn from_conversion(conversion: MessageHubProtoError) -> Self {
+        Self::ConversionError(conversion)
+    }
+}
+
+impl<S> StoreCapabilities for GrpcStore<S>
+where
+    S: StoreCapabilities,
+{
+    type Retention = S::Retention;
+
+    fn get_retention(&self) -> Option<&Self::Retention> {
+        self.store.get_retention()
+    }
+}
+
+/// We implement the PropertyStore to override the behavior when retrieving or storing
+/// device owned properies. We only want to store and load from the store server owned properties.
+/// Device owned properties wont be stored and will be requested to the message hub server.
+impl<S> PropertyStore for GrpcStore<S>
+where
+    S: PropertyStore,
+    S::Err: Display,
+{
+    type Err = GrpcStoreError<S::Err>;
+
+    async fn store_prop(&self, prop: StoredProp<&str, &AstarteType>) -> Result<(), Self::Err> {
+        if prop.ownership.is_server() {
+            self.store
+                .store_prop(prop)
+                .await
+                .map_err(GrpcStoreError::from_err)
+        } else {
+            Ok(())
+        }
```

Comment:
you could increase readability with:
```suggestion
// we only store server-owned properties
        if prop.ownership.is_device() { // or use   !prop.ownership.is_server()
            return Ok(());
        }

        self.store
            .store_prop(prop)
            .await
            .map_err(GrpcStoreError::from_err)
```

Reply:
the same stands for all the other methods where the else condition returns Ok(())